Sources/TiqrCoreObjC/Classes/IdentityService.m

```diff
@@ -71,6 +71,28 @@ - (void)deleteIdentityProvider:(IdentityProvider *)identityProvider {
     [self deleteObject:identityProvider];
 }
 
+- (void)deleteAllIdentitiesAndProviders: (NSError *)error {
+    // Delete the identities first
+    NSFetchRequest *identityRequest = [[NSFetchRequest alloc] initWithEntityName:@"Identity"];
+    NSBatchDeleteRequest *deleteIndentities = [[NSBatchDeleteRequest alloc] initWithFetchRequest:identityRequest];
+
+    [self.persistentStoreCoordinator executeRequest: deleteIndentities
+                                   withContext: self.managedObjectContext
+                                         error: &error];
+    if (error != nil) {
+        return;
+    }
+    
```

Comment:
If the relationship deletion rules of Core Data are setup correctly (big if) you would only need one deletion rule, which would then trigger the deletion of referenced objects.

Reply:
You are right, I tested this and only deleting the identities was enough